src/llvm-julia-licm.cpp

```diff
@@ -275,6 +274,17 @@ struct JuliaLICM : public JuliaPassContext {
                     if (valid) {
                         ++HoistedAllocation;
                         moveInstructionBefore(*call, *preheader->getTerminator(), MSSAU, SE);
+                        IRBuilder<> builder(preheader->getTerminator());
+                        auto obj_i8 = builder.CreateBitCast(call, Type::getInt8PtrTy(call->getContext(), call->getType()->getPointerAddressSpace()));
+                        // Note that this alignment is assuming the GC allocates 16-byte aligned memory
```

Comment:
It doesn't always. Small objects (<16 bytes) are only pointer-aligned.

Reply:
So we can assume alignment is at least `M.getDataLayout().getIntPtrType().getPrimitiveSizeInBits()`?

Reply:
It must be a power of 2, but yes, we have to assume someone might need to store an object of that alignment, up to our maximum alignment.